WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -6,77 +6,136 @@ import {
+  );
 
-  if (loading) return <Loading />;
-  if (error) return <Errored />;
+  const canEditTeamOfRole = (role) => role.group.group_type !== 'delegate_regions' && (
+    !!currentUserRoles.admin || currentUserRoles[role.group.name]?.status === 'leader'
+  );
+
+  if (
+    roleListFetch.loading || loggedInUserFetch.loading || saving || currentUserRoles === null
+  ) return <Loading />;
+  if (roleListFetch.error || loggedInUserFetch.error || error) return <Errored />;
 
-  return (data.activeRoles.length > 0
-    ? (
-      <>
-        <Header>Active Roles</Header>
-        <List>
-          <List.Item>
-            <RoleFormModal
-              trigger={(
-                <Button.Group basic vertical>
-                  <Button>Delegate</Button>
-                </Button.Group>
-          )}
-              userId={userId}
-              isActiveRole
-              sync={sync}
-            />
+  return (
+    <>
+      {activeNonHiddenRoles.length > 0
+        ? (
+          <>
+            <Header>Active Roles</Header>
+            <List divided relaxed>
+              {activeNonHiddenRoles.map((role) => (
+                <List.Item>
+                  <List.Icon
+                    name="edit"
+                    size="large"
+                    verticalAlign="middle"
+                    link
+                    disabled={!canEditRole(role)}
+                    onClick={() => setOpen(true)}
+                  />
+                  <List.Content>
+                    {canEditTeamOfRole(role) && (
+                    <List.Header as="a" href={`${teamUrl(role.group.id)}/edit`}>
+                      {`${I18n.t(`enums.user.role_status.${role.status}`)}, ${role.group.name}`}
+                    </List.Header>
+                    )}
+                    {!canEditTeamOfRole(role) && (
+                    <List.Header>
+                      {`${I18n.t(`enums.user.role_status.${role.status}`)}, ${role.group.name}`}
+                    </List.Header>
+                    )}
+                    {!!role.start_date && (
+                    <List.Description>
+                      {`Since ${role.start_date}`}
+                    </List.Description>
+                    )}
+                  </List.Content>
+                </List.Item>
+              ))}
+            </List>
+          </>
+        ) : (
+          <p>No Active Delegate Roles...</p>
+        )}
+      <Button onClick={() => setOpen(true)} disabled={isDelegate}>New Role</Button>
 
-          </List.Item>
-        </List>
-      </>
-    ) : (
-      <>
-        <p>No Active Roles...</p>
-        <RoleFormModal
-          trigger={<Button>New Role</Button>}
-          userId={userId}
-          isActiveRole={false}
-          sync={sync}
-        />
-      </>
-    )
+      <Modal
+        size="fullscreen"
+        onClose={() => {
+          setOpen(false);
+          sync();
+          window.location.reload(); // TODO: This is a hack to force the page to reload after
```

Comment:
Actually not to see updates, we can see the updates without even reloading. But one other tab will be visible only if the user is delegate. I've made the comment more elaborate so that it is more easier to understand now.